### PR TITLE
BUG: Bug in setitem raising ValueError when setting more than one column via array

### PR DESCRIPTION
**Pull request by phofl**
- [x] xref #37954
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Did not add whatsnew, cause works on 1.1.4

Array to set has to have more or less rows than columns, otherwise bug won't occur. Wrong values would be set silently here.

This is only a regression on master, not related to any release

cc @jbrockmendel

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -667,6 +667,8 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None):
                 if k not in self.obj:
                     if value is None:
                         self.obj[k] = np.nan
+                    elif is_array_like(value) and value.ndim == 2:
+                        self.obj[k] = value[:, i]
```
Comment (jbrockmendel) [inline pandas/core/indexing.py:671]: is this depending on having `len(self.obj) == len(value)`?

Comment (phofl) [reply pandas/core/indexing.py:671]: No this validation is done in this setitem

**Review by jreback — APPROVED** [review]
lgtm. cc @jbrockmendel

File: pandas/core/indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexing.py:671]: comment here pointing back to this PR, can be follow-up

**Comment by jbrockmendel** [comment]
does this address the perf complaint in the OP from the linked issue?

**Comment by jbrockmendel** [comment]
small comments, LGTM

**Comment by phofl** [comment]
No, only the regression raising an error.

File: pandas/core/indexing.py (hunk 1 above)
Comment (phofl) [reply pandas/core/indexing.py:671]: Done

**Comment by jbrockmendel** [comment]
LGTM cc @jreback (travis failure is unrelated)

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**